extension/iceberg/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+if (WIN32)
+    set(DuckDB_USE_STATIC_LIBS ON)
+else ()
+    set(DuckDB_USE_STATIC_LIBS OFF)
+endif ()
+find_package(DuckDB REQUIRED)
+
+include_directories(
+        ${PROJECT_SOURCE_DIR}/src/include
+        src/include
+        ${PROJECT_SOURCE_DIR}/extension/duckdb/src/include
+        ${PROJECT_SOURCE_DIR}/extension/httpfs/src/include
+        ${PROJECT_SOURCE_DIR}/extension/delta/src/include
+        ${DuckDB_INCLUDE_DIRS}
+)
+
+add_subdirectory(src/function)
+
+add_library(iceberg_extension
+        SHARED
+        ${ICEBERG_OBJECT_FILES}
+        src/iceberg_extension.cpp
+        ${PROJECT_SOURCE_DIR}/extension/delta/src/connector/delta_connector.cpp
+        ${PROJECT_SOURCE_DIR}/extension/duckdb/src/connector/duckdb_secret_manager.cpp
+        ${PROJECT_SOURCE_DIR}/extension/duckdb/src/connector/duckdb_connector.cpp
+        ${PROJECT_SOURCE_DIR}/extension/httpfs/src/s3_download_options.cpp)
+
+set_extension_properties(iceberg_extension iceberg iceberg)
+
+target_link_libraries(iceberg_extension
+        PRIVATE
+        ${DuckDB_LIBRARIES})
+
+if (NOT WIN32)
```

Comment:
Let's compile iceberg_install inside the src/installer folder.
See delta_installer as an example.
Let's make the main cmakelists.txt clean.